owncloudData/src/main/java/com/owncloud/android/data/availableoffline/AvailableOfflineRepositoryImpl.kt

```diff
@@ -0,0 +1,37 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Abel García de Prada
+ * Copyright (C) 2022 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.data.availableoffline
+
+import com.owncloud.android.domain.availableoffline.AvailableOfflineRepository
+import com.owncloud.android.domain.availableoffline.model.AvailableOfflineStatus
+import com.owncloud.android.domain.files.FileRepository
+import com.owncloud.android.domain.files.model.OCFile
+
+class AvailableOfflineRepositoryImpl(
+    val fileRepository: FileRepository,
+) : AvailableOfflineRepository {
+    override fun setFileAsAvailableOffline(fileToSetAsAvailableOffline: OCFile) {
+        fileRepository.updateFileWithNewAvailableOfflineStatus(fileToSetAsAvailableOffline, AvailableOfflineStatus.AVAILABLE_OFFLINE)
+    }
+
+    override fun unsetFileAsAvailableOffline(fileToSetAsAvailableOffline: OCFile) {
+        fileRepository.updateFileWithNewAvailableOfflineStatus(fileToSetAsAvailableOffline, AvailableOfflineStatus.NOT_AVAILABLE_OFFLINE)
+    }
```

Comment:
Is it correct that these methods belonging to a repository invoke methods from other repositories? 🤔. If that's the case for every method, maybe we can instance a `FileRepository` in the usecases that need this operations instead of adding boilerplate code like this. What do you think?